### PR TITLE
Added OnPlayerSpawnFailed delegate to SpatialGameInstance

### PR DESCRIPTION
**Pull request by improbable-andreaskrugersen**
Added OnPlayerSpawnFailed delegate to SpatialGameInstance;
Made OnConnected and OnConnectionFailed BlueprintAssignable

#### Tests
Started a local deployment and deleted the SpatialSpawner to force the spawn request to fail.

#### Documentation
release note

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialGameInstance.h [hunk 1]
```diff
@@ -53,11 +54,17 @@ class SPATIALGDK_API USpatialGameInstance : public UGameInstance
 
 	void HandleOnConnected();
 	void HandleOnConnectionFailed(const FString& Reason);
+	void HandleOnPlayerSpawnFailed(const FString& Reason);
 
 	// Invoked when this worker has successfully connected to SpatialOS
+	UPROPERTY(BlueprintAssignable)
 	FOnConnectedEvent OnConnected;
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialGameInstance.h:61]: Raised by Miron in the other PR: we may want to change the names to `OnSpatialConnection` etc to be very clear that these will only be triggered when running on Spatial.

File: CHANGELOG.md [hunk 2]
```diff
@@ -73,6 +73,7 @@ Usage: `DeploymentLauncher createsim <project-name> <assembly-name> <target-depl
 - Added `OnClientOwnershipGained` and `OnClientOwnershipLost` events on Actors and ActorComponents. These events trigger when an Actor is added to or removed from the ownership hierarchy of a client's PlayerController.
 - You can now generate valid schema for classes that start with a leading digit. The generated schema class will be prefixed with `ZZ` internally.
 - Handover properties will be automatically replicated when required for load balancing. `bEnableHandover` is off by default.
+- Added `OnPlayerSpawnFailed` delegate to `SpatialGameInstance`. This is helpful if you have established a successful connection but the server worker crashed. `OnConnected` and `OnConnectionFailed` are now also blueprint-assignable.
```
Comment (joshuahuburn) [inline CHANGELOG.md:76]: `OnConnected` and `OnConnectionFailed` were also renamed and this is something our customers bind too. Could you please make another PR to either revert this or update the Changelog to reflect the new names.

Comment (improbable-andreaskrugersen) [reply CHANGELOG.md:76]: I opened https://github.com/spatialos/UnrealGDK/pull/2038
